core/field.js

```diff
@@ -422,6 +422,16 @@ Blockly.Field.prototype.toXml = function(fieldElement) {
  * @package
  */
 Blockly.Field.prototype.saveState = function() {
+  if (Blockly.Field.prototype.saveState === this.saveState &&
+      Blockly.Field.prototype.toXml !== this.toXml) {
+    var elem = Blockly.utils.xml.createElement("field");
+    elem.setAttribute("name", this.name || '');
+    var text = Blockly.Xml.domToText(this.toXml(elem));
+    return text.replace(
+        'xmlns="https://developers.google.com/blockly/xml" ', '');
+  }
+  // Either they called this on purpose from their saveState, or they have
+  // no implementations of either hook. Just do our thing.
```

Comment:
Implementation looks fine (here and, AFAICT, in the other files too), but it'd be nice if the common code could be factored out, leaving just a call and check like:
```JS
  var xml = this.saveLegacyState(Blockly.Field);
  if (xml) return xml;  // Subclass overrides toXml but not saveState.
```
in each class, using a using a (`@protected`)
```
Blockly.Field.prototype.saveLegacyState = function(superclass) {
  if (superclass.prototype.saveState !== this.saveState ||
      superclass.prototype.toXml === this.toXml) {
    return null;
  }
  var elem = Blockly.utils.xml.createElement("field");
  elem.setAttribute("name", this.name || '');
  var text = Blockly.Xml.domToText(this.toXml(elem));
  return text.replace(
        'xmlns="https://developers.google.com/blockly/xml" ', '');
}
```
(Maybe verify that `saveLegacyState` can't return the empty string!)

Reply:
That sounds perfect. I wanted to factor this out, but I wasn't sure how to do it without creating a static function. Thank you :D